### PR TITLE
New tfy agent

### PR DESCRIPTION
**Pull request by dunefro**

### CODE REVIEW

File: charts/tfy-agent/values.yaml [hunk 1]
```diff
@@ -1,123 +1,296 @@
...
+  rbac:
+    enabled: true
 
-image:
-  repository: truefoundrycloud/tfy-agent
-  pullPolicy: IfNotPresent
-  # Overrides the image tag whose default is the chart appVersion.
-  tag: "a9af8a51d8b0373f4d0c469119ed88a488c5709f"
+# configuration for tfy-agent and tfy-agent-proxy
+config:
```
Comment (debajyoti-truefoundry) [inline charts/tfy-agent/values.yaml:11]: Normally these are done via the `.env.default` approach.

https://github.com/truefoundry/tfy-agent/blob/main/.env.default
https://github.com/truefoundry/helm-charts/blob/fd9a78ec2a7fc3b21f18cc610f1eff6d591b478c/charts/tfy-agent/values.yaml#L58

Going forward, do we need to also add a config field, if we introduce an env var?

File: charts/tfy-agent/values.yaml [hunk 2]
```diff
@@ -1,123 +1,296 @@
...
-  liveness:
-    port: 3000
-    path: /
 
-nodeSelector: {}
-tolerations: {}
-affinity: {}
+  ## Number of replicas of tfyAgent
+  #
+  replicaCount: 1
```
Comment (debajyoti-truefoundry) [inline charts/tfy-agent/values.yaml:202]: We should not expose this option for `tfy-agent`/`tfy-agent-proxy` as of now. There should be strictly only 1 pod for these services.

Comment (debajyoti-truefoundry) [reply charts/tfy-agent/values.yaml:202]: Later if we want to implement a hot stand-by pod via leader election, we can expose this.

File: charts/tfy-agent/values.yaml [hunk 3]
```diff
@@ -1,123 +1,296 @@
...
 
-envSecretName: tfy-agent-env-secret
-podAnnotations: {}
+  ## Image of tfyAgent
+  ##
+  image:
+    repository: truefoundrycloud/tfy-agent
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: "a9af8a51d8b0373f4d0c469119ed88a488c5709f"
```
Comment (debajyoti-truefoundry) [inline charts/tfy-agent/values.yaml:114]: Is this going to be automatically updated via our existing CI system? Or will this be manual?

https://github.com/truefoundry/tfy-agent/blob/a9af8a51d8b0373f4d0c469119ed88a488c5709f/.github/workflows/build-n-publish.yaml#L61

File: charts/tfy-agent/values.yaml [hunk 4]
```diff
@@ -1,123 +1,296 @@
...
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+# Enabling virtual services for istio
+# disabled by default
+# required when tfy-agent is to be exposed outside
```
Comment (debajyoti-truefoundry) [inline charts/tfy-agent/values.yaml:290]: We will not need to expose Agent outside. In the current architecture, we should ensure that the Agent is not contactable from the outside cluster apart form our control-plane, which does not require a VS.

File: charts/tfy-agent/Chart.yaml [hunk 5]
```diff
@@ -1,6 +1,24 @@
 apiVersion: v2
 name: tfy-agent
+description: A Helm chart for Kubernetes
```
Comment (srihari-tf) [inline charts/tfy-agent/Chart.yaml:3]: this is a placeholder description?

File: charts/tfy-agent/values.yaml (hunk 1 above)
Comment (dunefro) [reply charts/tfy-agent/values.yaml:11]: We will introduce `extraEnvVars: []` feature for both applications in coming release.

**Review by debajyoti-truefoundry — APPROVED** [review]

File: charts/tfy-agent/values.yaml [hunk 6]
```diff
@@ -1,123 +1,304 @@
...
-  tag: "a9af8a51d8b0373f4d0c469119ed88a488c5709f"
+# configuration for tfy-agent and tfy-agent-proxy
+config:
+  # token to connect to the contorl plane
+  clusterToken: ""
+
+  # name of the secret containing the cluster token
+  # the key must be token
+  # if this is set it will override config.clusterToken
+  clusterTokenSecret: ""
```
Comment (debajyoti-truefoundry) [inline charts/tfy-agent/values.yaml:18]: What should be the key name within the secret?

Comment (srihari-tf) [inline charts/tfy-agent/values.yaml:18]: if user provides a `clusterTokenSecret` the expectation might be that it already exists in the cluster. But current behaviour seems to be to create a new secret even when `clusterTokenSecret` is provided.

File: charts/tfy-agent/templates/_helpers.tpl [hunk 7]
```diff
@@ -24,106 +24,75 @@ Expand the name of the chart.
...
-{{- if and $val (contains "${k8s-secret" ($val | toString)) }}
-- name: {{ $key }}
-  valueFrom:
-    secretKeyRef:
-      name: {{ $.Values.envSecretName }}
-      key: {{ $val | trimPrefix "${k8s-secret/" | trimSuffix "}" }}
+Create the name of the secret which will contain cluster token
+*/}}
+{{- define "tfy-agent.secretName" -}}
+{{- if .Values.config.clusterTokenSecret }}
```
Comment (srihari-tf) [inline charts/tfy-agent/templates/_helpers.tpl:93]: we can skip this condition now right?

**Pull request merged**